contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -453,6 +448,9 @@ contract OCR2DRRegistry is
     }
     s_subscriptions[commitment.subscriptionId].balance -= bill.totalCost;
     // Pay out signers their portion of the DON fee
+    // Paying all signers will significantly increase gas usage, especially as DON size increases.
```

Comment:
How significantly? It seems like it's only one map value increment per node, right?

Reply:
While I am not exactly sure of the total, I would guess >6,000 gas per node paid. (800 for an SLOAD + 5,000 for an SSTORE + computation for checking for sufficient balance)

Reply:
Two justifications:
- faulty nodes do not get paid because they did not sign
- if DON fees become dynamic, such as when charging for data usage, this may be important

We can add an extra check to skip this logic if signerPayment = 0

I'd like for us to come up with hard numbers on the gas difference and then we can discuss the tradeoffs.